engine/src/main/java/org/hibernate/validator/internal/engine/ValidationContext.java

```diff
@@ -238,15 +220,15 @@ public ConstraintValidatorManager getConstraintValidatorManager() {
 	 * created for parameter validation, {@code null} otherwise.
 	 */
 	public List<String> getParameterNames() {
-		if ( parameterNameProvider == null ) {
+		if ( !ValidationOperation.PARAMETER_VALIDATION.equals( validationOperation ) ) {
```

Comment:
Or even raise an exception? It seems calling this method in this case would indicate a programming error?

Reply:
+1

Reply:
This is, kind of, related to suggestion to split the `ValidationContext` into multiple ones (https://hibernate.atlassian.net/browse/HV-1526). The way how this method is used:
```java
// create a constraint validator context
ConstraintValidatorContextImpl constraintValidatorContext = new ConstraintValidatorContextImpl(
		validationContext.getParameterNames(),
		validationContext.getClockProvider(),
		valueContext.getPropertyPath(),
		descriptor
);
```
wouldn't allow to just throw an exception here. So I was thinking that if we create different kinds of `ValidationContext`s we could then also have different implementations of `HibernateConstraintValidatorContext` and then we wouldn't need such kind of checks here, as well as in `ValidationContext#createConstraintViolation()`.